### PR TITLE
CNV-35729: Improve modal save while there are changes occurred in the back-end - part 2 network

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

Changing from using k8sUpdate to K8sPatch as it can cause the promise to fail if the VM has changed

### CODE REVIEW

File: src/utils/components/NetworkInterfaceModal/utils/helpers.ts [hunk 1]
```diff
@@ -35,6 +37,28 @@ export const updateVMNetworkInterfaces = (
   return updatedVM;
 };
 
+export const patchVMNetworkInterfaces = (
```
Comment (metalice) [inline src/utils/components/NetworkInterfaceModal/utils/helpers.ts:40]: Please change the function name to updateVM...

File: src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkInterfaceActions.tsx [hunk 2]
```diff
@@ -72,17 +60,17 @@ const NetworkInterfaceActions: FC<NetworkInterfaceActionsProps> = ({
...
-          })
-        }
+        onSubmit={() => {
+          const isHotPlug = Boolean(nicPresentation?.iface?.bridge);
+          const networks = isHotPlug
+            ? getNetworks(vm)
+            : getNetworks(vm)?.filter(({ name }) => name !== nicName);
+          const interfaces = updateInterfacesForDeletion(isHotPlug, nicName, vm);
+
+          return patchVMNetworkInterfaces(vm, networks, interfaces);
```
Comment (metalice) [inline src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkInterfaceActions.tsx:70]: can this be exported to util funciton?

Comment (avivtur) [reply src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkInterfaceActions.tsx]: Sure done

File: src/utils/components/NetworkInterfaceModal/utils/helpers.ts (hunk 1 above)
Comment (avivtur) [reply src/utils/components/NetworkInterfaceModal/utils/helpers.ts]: Sure done

**Pull request merged**